lib/topic_view.rb

```diff
@@ -135,6 +135,8 @@ def initialize(topic_or_topic_id, user = nil, options = {})
     @filtered_posts = apply_default_scope(@filtered_posts)
     filter_posts(options)
 
+    @post_user_badges = UserBadge.for_post_header_badges(@posts) if @posts
```

Comment:
I think we need to consider the `SiteSetting.enable_badges && SiteSetting.show_badges_in_post_header` site settings here. We don't want to be eager loading when we don't need to. I think what we can do something like this for the `post_user_badges` method?

```
def post_user_badeges
  @post_user_badges ||= begin
    UserBadge.for_post_header_badges(@posts).reduce({}) do |hash, user_badge|
      hash[user_badge.post_id] ||= []
      hash[user_badge.post_id] << user_badge
      hash
    end
  end
end
```

We memoize the results so that the query is only triggered once even if the method is called multiple times from different instance of `PostSerializer`. We also return a hash instead so that the look up from the `PostSerializer` does not involve looping through all the user badges for all the posts each time.